### PR TITLE
Add VAF stripplot

### LINKED ISSUE #208: add VAF strip plot
As done here https://github.com/uclahs-cds/project-disease-KidneyTumor-GHLR-000108-SubclonalArchitecture/blob/swu-sSNV-callers-QC-plotting/sSNV-analysis/plot_stripplot_adjVAF_combo_sSNVcallers.R

### PR DESCRIPTION
**Pull request by yashpatel6**
# Description

### Closes #208 

Adding a small workflow to generate a stripplot for the VAFs. Example plot (with A-mini):
image

## Testing Results
Tested with NFTest:
- `a_mini-all-tools-std-input`: `/hot/software/pipeline/pipeline-call-sSNV/Nextflow/development/unreleased/yashpatel-add-vaf-strip-plot/log-nftest-20240911T234607Z.log`
- `a_mini-all-tools-vcf-input`: `/hot/software/pipeline/pipeline-call-sSNV/Nextflow/development/unreleased/yashpatel-add-vaf-strip-plot/log-nftest-20240912T174040Z.log`

# Checklist

- [X] I have read the [code review guidelines](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3187646/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List).

- [X] I have reviewed the [Nextflow pipeline standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3193890/Nextflow+pipeline+standardization).

- [X] The name of the branch is meaningful and well formatted following the [standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List), using \[AD_username (or 5 letters of AD if AD is too long)]-\[brief_description_of_branch].

- [X] I have set up or verified the branch protection rule following the [github standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3190380/GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.

- [X] I have added my name to the contributors listings in the ``manifest`` block in the `nextflow.config` as part of this pull request; I am listed already, or do not wish to be listed. (*This acknowledgement is optional.*)

- [X] I have added the changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

- [ ] I have updated the version number in the `metadata.yaml` and `manifest` block of the `nextflow.config` file following [semver](https://semver.org/), or the version number has already been updated. (*Leave it unchecked if you are unsure about new version number and discuss it with the infrastructure team in this PR.*)

- [X] I have tested the pipeline on at least one A-mini sample.

### CODE REVIEW

File: script/prepare-VAFs.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    elif 'DP4' in variant_data_keys:
+        raw_vaf = get_vaf_somaticsniper(variant, sample)
+    elif all(base+'U' in variant_data_keys for base in ['A', 'C', 'G', 'T']):
+        raw_vaf = get_vaf_strelka2(variant, sample)
+    else:
+        raise ValueError(
+            f'Unable to identify variant source for: {variant} with {variant_data_keys}'
+        )
+
+    return raw_vaf * purity
```
Comment (sorelfitzgibbon) [inline script/prepare-VAFs.py:159]: Am i correct that the VAF for any particular variant is extracted from any one of the input VCFs (based on order here)?   The different tools don't all provide the same numbers, `Mutect2` seems to provide filtered read count, where the others provide unfiltered or both (`Strelka2`).   Is this accounted for in any way?  I'm not sure there is a simple way.

**Review by sorelfitzgibbon — APPROVED** [review]
Looks good to me.  Just a few non-blocking comments.

**Comment by sorelfitzgibbon** [comment]
I ran this on a set of PCAWG wgs VCFs:
`/hot/code/sfitzgibbon/gitHub/uclahs-cds/pipeline-call-sSNV/test/yaml/DO8264-vcfs.yaml`
`/hot/code/sfitzgibbon/gitHub/uclahs-cds/pipeline-call-sSNV/test/config/a_mini-all-tools.config`
output: `/hot/user/sfitzgibbon/PCAWG-63/analysis/yashpatel-add-vaf-strip-plot`
image

File: script/prepare-VAFs.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+def get_vaf_mutect2(variant: dict, sample: str) -> Tuple[int, int]:
+    """ Calculate read count for Mutect2 format """
+    sample_index = variant.samples.index(sample)
+    ref_reads = int(variant.samples[sample_index]["AD"][0])
+    alt_reads = int(variant.samples[sample_index]["AD"][1])
+    total_reads = ref_reads + alt_reads
+
+    return alt_reads/total_reads
+
```
[the remark this answers is not in the document]
Comment (yashpatel6) [reply script/prepare-VAFs.py:127]: I lifted these functions from SRC-util so I left the names as such, I'm not entirely sure how I would name it so I'll leave it as is for now

File: script/prepare-VAFs.py (hunk 1 above)
Comment (yashpatel6) [reply script/prepare-VAFs.py:159]: This is correct, the VAF for all variants is extracted in a tool-specific manner, ie. there is a VAF extracted per variant per tool and then averaging steps are done to reconcile/combine the VAFs. So it's accounted for through averaging essentially

File: script/prepare-VAFs.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+def get_vaf_somaticsniper(variant: dict, sample: str) -> Tuple[int, int]:
+    """ Calculate read count for SomaticSniper format """
+    # using DP4
+    sample_index = variant.samples.index(sample)
+    ref_reads = int(variant.samples[sample_index]["DP4"][0]) \
+        + int(variant.samples[sample_index]["DP4"][1])
+    alt_reads = int(variant.samples[sample_index]["DP4"][2]) \
+        + int(variant.samples[sample_index]["DP4"][3])
```
[the remark this answers is not in the document]
Comment (yashpatel6) [reply script/prepare-VAFs.py:115]: That could be an option to update, I'm leaving it at the moment to keep it on parity with the SRC-util parsers

**Pull request merged**